aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -217,7 +232,41 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim, c10::option
   return result;
 }
 
+std::tuple<Tensor&, Tensor&> cummax_out(Tensor& values, Tensor& indices, const Tensor& self, int64_t dim) {
+  check_scalar_type_device_layout_equal(values, self);
+  check_scalar_type_device_layout_equal(indices, at::empty({0}, self.options().dtype(at::kLong)));
```

Comment:
nit: do you need the `{0}` here? Can't you give `{}` so that it does not actually allocate memory?